### DISCUSSION TITLE
Debate around improving and adding concrete use-case examples

### DISCUSSION THREAD

**Discussion by elribonazo** [hyperledger-identus/hyperledger-identus#81] [Ideas]
As part of our roadmap we are going to be improving our documentation by adding better code examples for concrete use-cases.

## How can I add an idea?
Add a reply to this thread with the content of your use-case example and what would you want to focus on that example.

## How do we vote on the proposals
Click in the Arrow up button next to each reply and the top proposals will be the ones we are going to prioritize and this will be an ongoing discussion.

## What to expect after 
We will be attaching in this github discussion the tickets on each respective repository where we are going to be working on that example.
Creating specific workshops, etc

**Comment by elribonazo** [comment]
I will propose to build an interactive demo with JWT and another with SDJWT. 

### Making the demos interactive
In order to make our demos more interactive we will:
1. Easy to start from a sh script, all is built and configured through this SH script.
2. Configure everything inside the Demo portal
       2.1 Cloud Agent endpoint
       2.2 Edge Agent configuration
3. Demo wizard, that will step by step explain what is happening with code examples on each step
       3.1 API calls example + copy Curl button
       2.2 Show code examples
       
       
### Content of the Demo:
1. Demo configuration wizard: Configuration screen where the user can configure the demo, endpoints, storage, mediator, etc
2. Deployment dashboard: Shows you in a screen which services are running and an explanation of what this services do, a link to the exposes endpoints, etc
3. JWT or SDJWT Credential flows: Shows a step by step detailed Issuance and Presentation flow for the chosen credential type, all with code examples easy to copy paste, etc.

**Comment by FabioPinheiro** [comment]
The ticket https://github.com/hyperledger/identus/issues/77 is partially related with this discussion

**Comment by incubiq** [comment]
A full sample with AnonCreds would be great (including schema / definitions)

**Comment by incubiq** [comment]
Just in case you want to access some examples of what can be done with Identus:

 - Identus proof of capability video (a review of Identus capabilities including issuing Entity, Wallet, DID, VC, and Proof)  : https://youtu.be/4DyPuZr_3PA

 - An open src Cloud Identity Agent (in nodeJS), making use of Identus : https://github.com/incubiq/ai_identity ; here you  also have a "how to" for installing Identus on AWS

 - My API doc for the Cloud Identity Agent: https://incubiq.gitbook.io/opensourceais/identity-agent-lib

 - My final closing video report for the AI Identity project, which showcases use of Identus in a case of putting Identity on AIs: https://youtu.be/RL9bolGYcEU

**Reply by FabioPinheiro** [reply]
I would love to have a list of examples like this in the documentation.
I think we need to create a section just for examples.

**Reply by FabioPinheiro** [reply]
Here is the list of Community-Projects https://github.com/hyperledger/identus/wiki/Community-Projects. We need to was this one.

**Reply by FabioPinheiro** [reply]
Done

**Comment by mixmix** [comment]
# How to set up a Cloud Agent

## Level 1 : minimal path to get a demo spun up 
  
"I'm new to Identus and just wanna see if it work"
"I want to see all the credentials flows working, maybe write some integration tests"

Mix: I'm currently at this level. I wished there was a guide written for just this "get it off the ground/ prove it works" path. For me it felt like the amount needed to get over this initial hump to even be confident the parts play together was quite a lot. 

## Level 2 : production setup

"I want to set up a safe Cloud Agent"

Mix: There are so many options and I don't know what's actually needed, what is nice to have, and actually what best practices are for servers like this. What should I be considering for backups? What is a sensible default for safety? What configuration MUST I change from the default?

Ideally a guide for people who can code with SDKs but are not DevOps / SRE specialists (me!)

**Comment by FabioPinheiro** [comment]
# Trustify showcase

The Trustify verification portal from the EduDID

At the first one seems it there are some similarities with the verification portal that we have on Identus/Atala version 1.x, but improved and much nicer, with new features.

IMO seems a nice showcase. What do you guys think?

**Reply by FabioPinheiro** [reply]
@danniebgoode Can you ask if the showcase video is public?

**Reply by DannieBGoode** [reply]
it's not public
